---
title: "R Notebook"
output: html_notebook
---
---
title: "GeumCultureProject"
author: "Kacey Lange"
date: "2024-07-17"
output: html_document
---
# Libraries
  
followed instructions: https://github.com/bioc/sangeranalyseR

``` {r}
library(ggplot2)
library(readxl)
library(tidyverse)
library(googlesheets4)
library(dplyr)
library(tidyr)
library(broom)
library(purrr)
#library(vegan)
library(RColorBrewer)
library(forcats)
#library(boxr)
#if (!requireNamespace("BiocManager", quietly = TRUE))
    #install.packages("BiocManager")

# The following initializes usage of Bioc devel
#BiocManager::install(version='devel')

#BiocManager::install("sangeranalyseR")
#install.packages("devtools")
#BiocManager::install("GenomeInfoDbData")
BiocManager::install("GenomeInfoDb")

library(devtools)

## Install the release version
#install_github("roblanf/sangeranalyseR", ref = "master")

## Install the development version
#install_github("roblanf/sangeranalyseR", ref = "develop")
#install.packages("GenomeInfoDbData")
library(sangeranalyseR) #go to GitHub for TMZ file or
#install(BiocManager)
library(sangerseqR)
#library(Biostrings)
#library(ggtree)
library(viridisLite)
#library(treeio)
#library(geiger)
library(ape)
#library(ggnewscale)
library(phytools)

#devtools::install_github("brendanf/FUNGuildR")

library(FUNGuildR)


library(remotes)
library(ape)

#BiocManager::install("ggtree")


library(picante)
library(phyloseq)
library(nlme)
#library(FUNGuildR)

#BiocManager::install("dada2")
library(dada2)
library(ggmap)

```

# Sanger Sequencing data ---

Following is for documention. If you need to work with alignments,
McKenzie recommends using the "contigs_object.RData" file on GDrive. You
need to use `sangeranalyseR` package in order to work with it.

load("C:/Users/Kacey/Documents/R/Projects/Culture/contigs_object.RData")

Genewiz seq data folder id (ancestor folder id): 112149094632

## there are two different naming conventions for ab1 files for some reason (anas vs mine)
``` {r}
names <- as.vector(list.files('C:/Users/Kacey/Documents/Tulane/Lab/Culture/Geum roots/Data/Trace/Pass'))

str_replace_all(names,'1F','F1') %>%
  as.vector() -> replace

setwd("C:/Users/Kacey/Documents/Tulane/Lab/Culture/Geum roots/Data/Trace/Pass")
file.rename(from = names, to=replace)
```

## Align sequences and write to FASTA ---

McKenzie:
Sangeranalyser defaults to Mott (M1 method) Phred quality cutoff = 30(.001 on a log scale). default to 2 reads necessary to build contig(i.e. F+R); 20 is default min length of read after trimming; Mott wasbad b/c we have a lot of random low quality bases interspersed.
*Used trimmomatic algorithm for sliding window (instead of Mott)* so that a single low quality base doesnt terminate sequence too early.this is a common problem in this data set.
also tried using reference amino acid sequence, but performed much worsethan normal alignment process consensus amino based on M2 cutoff = 30,sliding window =10
"FFLLSSSSYY\*\*CC\*WLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG"

Emily:
The M2 method was bad because there was a lot of uncalled basepairs (i.e W,K,Y,...). With the M1 method, there were much less uncalled basepairs. Use trimming cutoff of 1e-04

Kacey:
Emily was right and the M1 method was better. I will continue with using M1.To properly make contigs, the samples have to have the same amount of characters. For example, I had to change sample 1-ITSF1.ab1 to 001-ITSF1.ab1 because I have samples with three characters (250-ITSF1.ab1). Without making them the same amount of characters, it will read multiple samples under 1 contig in the LaunchApp. I don't know if this changes the data or not? Update: it does change the data and fixing the characters loads the LaunchApp correctly and later on, assigns the guild better. 

This generates a new FASTA:

```{r, eval=FALSE}
contigs <- SangerAlignment(ABIF_Directory     = "C:/Users/Kacey/Documents/Tulane/Lab/Culture/Geum roots/Data/Trace/Pass",
                          inputSource         = "ABIF",
                          processMethod       = "REGEX",
                          REGEX_SuffixForward = "-ITSF1.ab1$",
                          REGEX_SuffixReverse = "-LR3.ab1$",
                         TrimmingMethod       = "M1",
                         M1TrimmingCutoff     = 1e-04,
                         M2CutoffQualityScore = NULL,
                         M2SlidingWindowSize  = NULL)
  
#contigs <- updateQualityParam(contigs,
                              #TrimmingMethod = "M2",
                              #M2CutoffQualityScore = 30,
                              #M2SlidingWindowSize  = 10)

writeFastaSA(contigs,
             outputDir         = "C:/Users/Kacey/Documents/Tulane/Lab/Culture/Geum roots/Data/FASTAS",
             compress          = FALSE,
             selection         = "contigs_unalignment")

launchApp(contigs)

```

# T-BAS

Mckenzie:
-take output FASTA (unaligned) and pass through NCSU ITSx to separate various loci of interest (should be ITS1, 5.8S, ITS2, some of LSU)
-Retain all metadata for OTU members
-ITS locus is included (first FASTA) - filter unknowns to selected taxon and generate UNITE report
-Run RDP with UNITE (ITS), FunGuild, and NCBI WWW bastn
-think I want to classify using Warcup b/c it seems to perform better athigher resolution taxon assignments 
(there is a citation for this somewhere)
-probably don't cluster on only one locus bc that would be dumb/probably
-actually skip clustering altogether bc we want info for each isolate
-Create OTU fasta files
-definitely don't do de novo tree construction, use Evolutionary placement algorithm (bc that will incorporate extant info from TBAS which is the whole point); Emily used de novo though?
EPA and EPA ng are very similar, basically differ in computing
efficiency.
-3/8/23 don't use RDP classifier b/c they only consider LSU and I don'tthink we actually have enough LSU coverage to solely depend on that

Kacey:
-take output FASTA (unaligned) and pass through NCSU ITSx to separate various loci of interest (should be ITS1, 5.8S, ITS2, some of LSU)
-DO NOT retain all metadata for OTU members. (my data showed the same OTU multiple times in the tree)
-ITS locus is included (first FASTA) - filter unknowns to selected taxon and generate UNITE report
-Run RDP with UNITE (ITS), and FunGuild
-Cluster on only one locus
-Cluster at 0.97
-Create OTU fasta files
-RAxML with de novo tree construction


*use result files from run QIZ4LVYB. This is good one.*

T-BAS data:
OTU is the otu name that was input for the isolate name in T-BAS, every isolate has a unique OTU.
Query.sequence is also unique to each isolate. It is the sample number (1-250).

```{r}
geumroot <- read.csv("C:/Users/Kacey/Documents/Tulane/Lab/Culture/Geum roots/T-BAS_Results/tbas21_archiveQIZ4LVYB_/assignments_report_addvoucherQIZ4LVYB.csv")

#query sequence 154 and 54 were OTU11
geumroot <- geumroot[-c(44,155),]
colnames(geumroot)[16]<- "otu" 
colnames(geumroot)[1]<- "Query.sequence"
geumroot$Query.sequence <- sprintf("%03d", geumroot$Query.sequence) #changing the numbers to 3 characters to match the contigs

```

## Phylogenetic Tree

TBAS tree with all sequences -> QIZ4LVYB.nwk
TBAS tree after selecting all query sequences and pruning the tree -> CM5WSVTT.nwk *this is what I want*

Change tip labels to oTU numbers 

```{r}
#newtree <- read.tree("C:/Users/Kacey/Documents/Tulane/Lab/Culture/Geum roots/T-BAS_Results/tbas21_archive3HXRGT5G_/3HXRGT5G.nwk")

trimmednwk <-read.tree("C:/Users/Kacey/Documents/Tulane/Lab/Culture/Geum roots/T-BAS_Results/tbas21_archiveCM5WSVTT_/CM5WSVTT.nwk.txt")

# tree with species names 
#trimmednwk$tip.label <- geumroot[[12]][match(trimmednwk$tip.label, geumroot[[1]])]
#taxtree <- ggtree(trimmednwk)+geom_tiplab(size=1)
#ggsave('C:/Users/Kacey/Documents/Tulane/Lab/Culture/Geum roots/taxtree.pdf')

# tree with OTUs 
trimmednwk$tip.label <- geumroot[[16]][match(trimmednwk$tip.label, geumroot[[1]])]
#trimmednwk <- ggtree(trimmednwk)+geom_tiplab(size=1)
#ggsave('C:/Users/Kacey/Documents/Tulane/Lab/Culture/Geum roots/otutree.pdf')

plot(trimmednwk)

cleanedtree <-read.tree("C:/Users/Kacey/Documents/Tulane/Lab/Culture/Geum roots/T-BAS_Results/tbas21_archiveCM5WSVTT_/CM5WSVTT.nwk.txt")

#tip 154 was a weird sample and was throiwng mpd data off so it is removed
dropped_tip_154 <- drop.tip(cleanedtree, tip= c('154'), trim.internal = TRUE, subtree = FALSE,
                          root.edge = 0, rooted = is.rooted(cleanedtree), collapse.singles = TRUE,
                          interactive = FALSE)

write.tree(dropped_tip_154, file = "C:/Users/Kacey/Documents/Tulane/Lab/Culture/Geum roots/Data/cleanedtree_V2.nwk", append = FALSE,
           digits = 5, tree.names = FALSE)

cleanedtree_v2 <- read.tree("C:/Users/Kacey/Documents/Tulane/Lab/Culture/Geum roots/Data/cleanedtree_V2.nwk")

cleanedtree_v2$tip.label <- geumroot[[16]][match(cleanedtree_v2$tip.label, geumroot[[1]])]

plot(cleanedtree_v2)
```

# Culture data

There are multiple OTUs for one species, so this is to clean it up

Culture IDs are unique to the sequence. It contains the site, community, and plant individual number.

```{r}
culture <- read_sheet("https://docs.google.com/spreadsheets/d/1Ys3msUUPmmOo1NeRqHfsdhJzU_mEwBOytZe6AmAribQ/edit?gid=0#gid=0", sheet = "Sheet1", na='NULL')
culture <- culture[-c(54,154),]

culture2<-culture%>%
  dplyr::select(Query.sequence,...3)%>%
  rename(PlantIndividual=...3)%>%
  separate(PlantIndividual, 
           into = c("Community", "PlantIndividual"), 
           sep = "(?<=[A-Za-z])(?=[0-9])"
           )
head(culture2)

culture2$Query.sequence <- sprintf("%03d", culture2$Query.sequence)

geumroot2<-geumroot%>%
  unite(CultureID,otu,Taxon.assignment,remove=F)%>%
  dplyr::select(Query.sequence,CultureID,Community_Type_QIZ4LVYB,Site_QIZ4LVYB,Latitude_QIZ4LVYB,Longitude_QIZ4LVYB,otu,Trophic.Mode,Guild,Taxon.assignment,GenusSpecies)%>%
  rename(Community=Community_Type_QIZ4LVYB,Site=Site_QIZ4LVYB,Lat=Latitude_QIZ4LVYB,Long=Longitude_QIZ4LVYB)%>%
mutate(Community = factor(Community, levels = c("SB", "MM", "DM", "FF")),Site = factor(Site, levels = c("P", "Q")))
head(geumroot2)

length(geumroot2$CultureID)

dat<-full_join(geumroot2,culture2,by="Query.sequence")%>%
  filter(is.na(otu)==F)
head(dat)
dim(dat)

#Collapse identical OTUs into one row. Looking at abundances of OTUs in the whole dataset. there are many many 1's (singletons). also this is needed to create the community dataset below
dat2<-dat%>%
  group_by(Community.x,Site,PlantIndividual,otu)%>%
  summarise(abundance=n())
as.data.frame(dat2)
data.frame(dat2$Community.x,dat2$Site,dat2$PlantIndividual,dat2$otu,dat2$abundance)

#make wide
dat3<-dat2%>%
  ungroup()%>%
  unite(CommunitySitePL,Community.x,Site,PlantIndividual,remove=F)%>%
  spread(otu,abundance,fill=0)
dat.comm<-data.frame(dat3[,5:37])
row.names(dat.comm)<-dat3$CommunitySitePL

```

## Faith's Phylogenetic distance

```{r}

PD<-pd(dat.comm,trimmednwk)

dat4<-data.frame(dat3[,1:4],PD,dat3[,5:37])
```

## MPD by CommunitySitePL

```{r}
phydist <- cophenetic(trimmednwk)
ses.mpd.result.notweighted <- ses.mpd(dat.comm, phydist, null.model="taxa.labels",abundance.weighted=FALSE, runs=999) #takes 5 min with 999
ses.mpd.result.notweighted
ses.mpd.result.notweighted$CommunitySitePL<-rownames(ses.mpd.result.notweighted)
ses.mpd.result.notweighted1<-ses.mpd.result.notweighted%>%
  select(CommunitySitePL,mpd.obs.z)%>%
  rename(mpd.obs.z.notweighted=mpd.obs.z)

ses.mpd.result.weighted <- ses.mpd(dat.comm, phydist, null.model="taxa.labels",abundance.weighted=TRUE, runs=999) #takes 5 min with 999
ses.mpd.result.weighted
ses.mpd.result.weighted$CommunitySitePL<-rownames(ses.mpd.result.weighted)
ses.mpd.result.weighted1<-ses.mpd.result.weighted%>%
  select(CommunitySitePL,mpd.obs.z)%>%
  rename(mpd.obs.z.weighted=mpd.obs.z)

dat5<-dat4%>%
  full_join(ses.mpd.result.notweighted1)%>%
  full_join(ses.mpd.result.weighted1)
  
  
#dat6<-data.frame(dat5[,1:6],dat5[,41:42],dat5[,7:40])
#head(dat6)

```

## MNTD

```{r}
phydist <- cophenetic(trimmednwk)
ses.mntd.result.notweighted <- ses.mntd(dat.comm, phydist, null.model="taxa.labels",abundance.weighted=FALSE, runs=999) #takes 5 min with 999
ses.mntd.result.notweighted
ses.mntd.result.notweighted$CommunitySitePL<-rownames(ses.mntd.result.notweighted)
ses.mntd.result.notweighted1<-ses.mntd.result.notweighted%>%
  select(CommunitySitePL,mntd.obs.z)%>%
  rename(mntd.obs.z.notweighted=mntd.obs.z)

ses.mntd.result.weighted <- ses.mntd(dat.comm, phydist, null.model="taxa.labels",abundance.weighted=TRUE, runs=999) #takes 5 min with 999
ses.mntd.result.weighted
ses.mntd.result.weighted$CommunitySitePL<-rownames(ses.mntd.result.weighted)
ses.mntd.result.weighted1<-ses.mntd.result.weighted%>%
  select(CommunitySitePL,mntd.obs.z)%>%
  rename(mntd.obs.z.weighted=mntd.obs.z)

dat5<-dat4%>%
  full_join(ses.mntd.result.notweighted1)%>%
  full_join(ses.mntd.result.weighted1)%>%
  full_join(ses.mpd.result.notweighted1)%>%
  full_join(ses.mpd.result.weighted1)
  
  
dat6<-data.frame(dat5[,1:6],dat5[,40:43],dat5[,7:39])
head(dat6)

```

## Phyloseq Object

```{r}
otus<-dat6[,11:43]
otus2<-t(otus)
sampleotus<-dat6[,c(1:10)]
taxonomyotus<-as.matrix(data.frame(Kingdom=row.names(otus2),Phylum=row.names(otus2),Class=row.names(otus2),Order=row.names(otus2),Class=row.names(otus2),Family=row.names(otus2),Genus=row.names(otus2),Species=row.names(otus2)))
rownames(taxonomyotus)<-row.names(otus2)

datp <- merge_phyloseq(otu_table(otus2,taxa_are_rows = T), tax_table(taxonomyotus), sample_data(sampleotus),trimmednwk)

#calculate unifrac distances
unifracp<-unifrac(otus,trimmednwk)
```

# Testing Blasting to UNITE

Make sure to get the most recently updated reference file from the UNITE database website.
https://doi.plutof.ut.ee/doi/10.15156/BIO/2959333
At the bottom under downloads

taxaonly2 <- read.csv("C:/Users/Kacey/Documents/Tulane/Lab/Culture/Geum roots/taxaonly.csv")

```{r}

unite.ref <- "C:/Users/Kacey/Documents/Tulane/Lab/Culture/UNITE/sh_general_release_dynamic_s_04.04.2024.fasta"


#I got this data from assignments_report_nodupsM76GMYAB
myotus<-read.csv("C:/Users/Kacey/Documents/Tulane/Lab/Culture/Geum roots/T-BAS_Results/tbas21_archiveQYR3BIB5_/assignments_report_nodupsQYR3BIB5.csv",header=T);rownames(myotus)<-myotus$abundance

#throwing out query sequence 154 and 54 (OTU11)
#myotus <- myotus[-c(2),]
myotus$Query.sequence <- sprintf("%03d", myotus$Query.sequence)

myotus2 <- myotus[,c(52,47)]
names(myotus2)[names(myotus2) == 'ITS'] <- 'sequence'
names(myotus2)[names(myotus2) == 'otu.1'] <- 'abundance'

#delete rows with no sequence
myotus2 <- myotus2[!(is.na(myotus2$sequence) | myotus2$sequence==""), ]

taxa<-assignTaxonomy(myotus2, unite.ref, multithread = TRUE, minBoot=70, tryRC = TRUE,outputBootstraps=T) #was minBoot=70
taxaonly<-data.frame(taxa$tax);rownames(taxaonly)<-1:dim(taxa$tax)[1]
taxaboot<-data.frame(taxa$boot);rownames(taxaboot)<-1:dim(taxa$tax)[1]
taxaonly
taxaboot
rownames(taxaonly)<-rownames(myotus)
genusspecies<-data.frame(otu=rownames(myotus),genusspecies=paste(gsub("^.*?__","",taxaonly[,"Genus"]),gsub("^.*?__","",taxaonly[,"Species"])),genusboot=taxaboot[,"Genus"],speciesboot=taxaboot[,"Species"])

sort(unique(genusspecies$genusspecies))
sort(unique(taxaonly$Phylum))

taxaonly%>%group_by(Phylum)%>%tally()
```

## FunGuild

using FUNGuildR

```{r}

#temp3 <- funguild_query("Buergenerula*", "taxon", db = fung)

#Doing it in terminal - use this
taxaonly2<-taxaonly
taxaonly2$taxonomy<-paste(taxaonly2$Kingdom,taxaonly2$Phylum,taxaonly2$Class,taxaonly2$Order,taxaonly2$Family,taxaonly2$Genus,taxaonly2$Species,sep=";")

write.csv(taxaonly2,"C:/Users/Kacey/Documents/Tulane/Lab/Culture/Geum roots/taxaonly.csv")

taxaonly2 <- read.csv("C:/Users/Kacey/Documents/Tulane/Lab/Culture/Geum roots/taxaonly.csv")

#the data frame that goes into the funguildR has to have a column named "Taxonomy" (has to be uppercase) and include your otu numbers too
taxaonly3<-data.frame(myotus2[,2],taxaonly2[,2:9])
colnames(taxaonly3)[1] <- "OTU"
colnames(taxaonly3)[9] <- "Taxonomy"
#taxaonly3$OTU <- stringr::str_replace(taxaonly3$OTU, '\\OTU', '')

guilds <- funguild_assign(taxaonly3)

guilds%>%filter(trophicMode==("Pathotroph"))
guilds%>%filter(trophicMode==("Symbiotroph"))

View(funguild_query("*Saprotroph*", "trophicMode", db = guilds))

#merge with OTU table (of datp) with guilds from above
otutomerge<-data.frame(otu_table(datp))
otutomerge$OTU<-rownames(otutomerge)
head(guilds)
guilds2<-guilds%>%
  full_join(otutomerge)
head(guilds2)
#throwing out query sequence 154 and 54 (OTU11)
#guilds2 <- guilds2[-c(2),]

#confident indicies 
ind<-which(guilds2$guild=="Plant Pathogen")
plantpathogen<-colSums(guilds2[ind,21:64])
plantpathogentaxa<-colSums(guilds2[ind,21:64]>0)
ind<-which(guilds2$trophicMode=="Pathotroph")
pathotroph<-colSums(guilds2[ind,21:64])
pathotrophtaxa<-colSums(guilds2[ind,21:64]>0)
ind<-which(guilds2$guild=="Ericoid Mycorrhizal")
ericoid<-colSums(guilds2[ind,21:64])
ericoidtaxa<-colSums(guilds2[ind,21:64]>0)
ind<-which(guilds2$trophicMode=="Symbiotroph")
symbiotroph<-colSums(guilds2[ind,21:64])
symbiotrophtaxa<-colSums(guilds2[ind,21:64]>0)
ind<-which(guilds2$trophicMode=="Saprotroph")
saprotroph<-colSums(guilds2[ind,21:64])
saprotrophtaxa<-colSums(guilds2[ind,21:64]>0)
ind<-which(guilds2$guild=="Plant Saprotroph")
plantsaprotroph<-colSums(guilds2[ind,21:64])
plantsaprotrophtaxa<-colSums(guilds2[ind,21:64]>0)
ind<-which(guilds2$guild=="Endophyte")
endophyte<-colSums(guilds2[ind,21:64])
endophytetaxa<-colSums(guilds2[ind,21:64]>0)

# not confident indicies
ind<-funguild_query("*Plant Pathogen*", "guild", db = guilds2)$OTU#trophicMode
ind2<-which(guilds2$OTU%in%ind)
plantpathogenbroad<-colSums(guilds2[ind2,21:64])
plantpathogenbroadtaxa<-colSums(guilds2[ind2,21:64]>0)
ind<-funguild_query("*Pathotroph*", "trophicMode", db = guilds2)$OTU#trophicMode
ind2<-which(guilds2$OTU%in%ind)
pathotrophbroad<-colSums(guilds2[ind2,21:64])
pathotrophbroadtaxa<-colSums(guilds2[ind2,21:64]>0)
ind<-funguild_query("*Ericoid Mycorrhizal*", "guild", db = guilds2)$OTU#trophicMode
ind2<-which(guilds2$OTU%in%ind)
ericoidbroad<-colSums(guilds2[ind2,21:64])
ericoidbroadtaxa<-colSums(guilds2[ind2,21:64]>0)
ind<-funguild_query("*Symbiotroph*", "trophicMode", db = guilds2)$OTU
ind2<-which(guilds2$OTU%in%ind)
symbiotrophbroad<-colSums(guilds2[ind2,21:64])
symbiotrophbroadtaxa<-colSums(guilds2[ind2,21:64]>0)
ind<-funguild_query("*Saprotroph*", "trophicMode", db = guilds2)$OTU#trophicMode
ind2<-which(guilds2$OTU%in%ind)
saprotrophbroad<-colSums(guilds2[ind2,21:64])
saprotrophbroadtaxa<-colSums(guilds2[ind2,21:64]>0)
ind<-funguild_query("*Plant Saprotroph*", "guild", db = guilds2)$OTU#trophicMode
ind2<-which(guilds2$OTU%in%ind)
plantsaprotrophbroad<-colSums(guilds2[ind2,21:64])
plantsaprotrophbroadtaxa<-colSums(guilds2[ind2,21:64]>0)
ind<-funguild_query("*Endophyte*", "guild", db = guilds2)$OTU#trophicMode
ind2<-which(guilds2$OTU%in%ind)
endophytebroad<-colSums(guilds2[ind2,21:64])
endophytebroadtaxa<-colSums(guilds2[ind2,21:64]>0)

#indicies with confidence ranking
ind<-funguild_query("*Plant Pathogen*", "guild", db = guilds2)#trophicMode
ind2<-ind[which(ind$confidenceRanking%in%c("Probable","Highly Probable")),"OTU"]
ind3<-which(guilds2$OTU%in%ind2)
plantpathogenbroadprobablehp<-colSums(guilds2[ind3,21:64])
plantpathogenbroadprobablehptaxa<-colSums(guilds2[ind3,21:64]>0)
ind<-funguild_query("*Pathotroph*", "trophicMode", db = guilds2)#trophicMode
ind2<-ind[which(ind$confidenceRanking%in%c("Probable","Highly Probable")),"OTU"]
ind3<-which(guilds2$OTU%in%ind2)
pathotrophbroadprobablehp<-colSums(guilds2[ind3,21:64])
pathotrophbroadprobablehptaxa<-colSums(guilds2[ind3,21:64]>0)
ind<-funguild_query("*Ericoid Mycorrhizal*", "guild", db = guilds2)#trophicMode
ind2<-ind[which(ind$confidenceRanking%in%c("Probable","Highly Probable")),"OTU"]
ind3<-which(guilds2$OTU%in%ind2)
ericoidbroadprobablehp<-colSums(guilds2[ind3,21:64])
ericoidbroadprobablehptaxa<-colSums(guilds2[ind3,21:64]>0)
ind<-funguild_query("*Symbiotroph*", "trophicMode", db = guilds2)
ind2<-ind[which(ind$confidenceRanking%in%c("Probable","Highly Probable")),"OTU"]
ind3<-which(guilds2$OTU%in%ind2)
symbiotrophbroadprobablehp<-colSums(guilds2[ind3,21:64])
symbiotrophbroadprobablehptaxa<-colSums(guilds2[ind3,21:64]>0)
ind<-funguild_query("*Saprotroph*", "trophicMode", db = guilds2)#trophicMode
ind2<-ind[which(ind$confidenceRanking%in%c("Probable","Highly Probable")),"OTU"]
ind3<-which(guilds2$OTU%in%ind2)
saprotrophbroadprobablehp<-colSums(guilds2[ind3,21:64])
saprotrophbroadprobablehptaxa<-colSums(guilds2[ind3,21:64]>0)
ind<-funguild_query("*Plant Saprotroph*", "guild", db = guilds2)#trophicMode
ind2<-ind[which(ind$confidenceRanking%in%c("Probable","Highly Probable")),"OTU"]
ind3<-which(guilds2$OTU%in%ind2)
plantsaprotrophbroadprobablehp<-colSums(guilds2[ind3,21:64])
plantsaprotrophbroadprobablehptaxa<-colSums(guilds2[ind3,21:64]>0)
ind<-funguild_query("*Endophyte*", "guild", db = guilds2)#trophicMode
ind2<-ind[which(ind$confidenceRanking%in%c("Probable","Highly Probable")),"OTU"]
ind3<-which(guilds2$OTU%in%ind2)
endophytebroadprobablehp<-colSums(guilds2[ind3,21:64])
endophytebroadprobablehptaxa<-colSums(guilds2[ind3,21:64]>0)

totalabundance<-colSums(guilds2[,21:64])

temp<-data.frame(sample_data(datp))
temp$rownames<-rownames(temp)
temp<-temp%>%
  dplyr::select(rownames,CommunitySitePL)
temp2<-data.frame(temp,
plantpathogen,plantpathogentaxa,plantpathogenbroad,plantpathogenbroadtaxa,plantpathogenbroadprobablehp,plantpathogenbroadprobablehptaxa,
pathotroph,pathotrophtaxa,pathotrophbroad,pathotrophbroadtaxa,pathotrophbroadprobablehp,pathotrophbroadprobablehptaxa,
ericoid,ericoidtaxa,ericoidbroad,ericoidbroadtaxa,ericoidbroadprobablehp,ericoidbroadprobablehptaxa,
symbiotroph,symbiotrophtaxa,symbiotrophbroad,symbiotrophbroadtaxa,symbiotrophbroadprobablehp,symbiotrophbroadprobablehptaxa,
saprotroph,saprotrophtaxa,saprotrophbroad,saprotrophbroadtaxa,saprotrophbroadprobablehp,saprotrophbroadprobablehptaxa,
plantsaprotroph,plantsaprotrophtaxa,plantsaprotrophbroad,plantsaprotrophbroadtaxa,plantsaprotrophbroadprobablehp,plantsaprotrophbroadprobablehptaxa,
endophyte,endophytetaxa,endophytebroad,endophytebroadtaxa,endophytebroadprobablehp,endophytebroadprobablehptaxa,
totalabundance)

dat7<-dat6%>%
  full_join(temp2)

write.csv(dat7,"C:/Users/Kacey/Documents/Tulane/Lab/Culture/Geum roots/GeumMastersheet.csv")

```

# Summary of files

```{r}
datp #phyloseq object
head(dat6) #big data frame, wide data format, dat3 plus PD and MPD data
dat7 #dat6 plus funguild pathogens/symbiotrophs
dat2 #long dataformat 
```

# Map

``` {r}

mapsheet <- read.csv("C:/Users/Kacey/Documents/Tulane/Lab/Culture/Geum roots/CultureMastersheet.csv")

Longitude <- mapsheet$DDLongitude
Latitude <- mapsheet$DDLatitude

register_stadiamaps(key ="d2f32a9d-6671-47c9-aae7-ccf6f1476c02")

smallmap <- get_stadiamap( bbox = c(left=-105.6, bottom =40.045, right =-105.57, top =40.065), maptype = "outdoors")


sm <- ggmap(smallmap) +
  geom_point(
      aes(Longitude, Latitude), data = mapsheet, 
    shape = 21, color = "gray25", fill = "yellow"
  )

bigmap <- get_stadiamap( bbox = c(left=-106, bottom =39.75, right =-105.25, top =40.25), maptype = "outdoors")


bm <- ggmap(bigmap) +
  geom_point(
      aes(Longitude, Latitude), data = mapsheet, 
    shape = 21, color = "gray25", fill = "yellow"
  )

```

# git hub token stuff

```{r}
install.packages("gitcreds")
library(gitcreds)
gitcreds::gitcreds_set()
 #first when it asks to enter password or token I put my computer password
 #then do gitcreds_set() again and select 2, then paste my token
#Note: use usename (email) and token, when RStudio wants the github password
```